### DISCUSSION TITLE
phonemes are different between dictionary and output

### DISCUSSION THREAD

**Discussion by cnlinxi** [espeak-ng/espeak-ng#1814] [Q&A]
Why the phoneme sequence in dictionary and the final output are different. For example, in Arabic:

**dictsource/ar_listx**:
![image](https://github.com/espeak-ng/espeak-ng/assets/11946496/babf9417-2332-4015-bdcc-a99399c41512)

but by the following command:

```shell
echo "دبي" | espeak-ng -qX -v ar
```

the output is:

```
Found: 'دبي' [dubajj]  
 d'ubajj
```

phoneme sequence has changed from `dubaj:` to `d'ubajj`.

What are the conversion rules? Where did the conversion occur?

Thanks a lot.

**Comment by jaacoppi** [comment]
That specific case is explained by the addition of stress. Probably in SetWordStress() in dictionary.c

The default stress rules are used if no explicit stress is set in *_list files.

**Reply by cnlinxi** [reply]
Thank you for your reply.
I want to know why phoneme has changed from `j:` to `jj`. I cannot find this map in `phsource/phonemes`.

**Comment by jaacoppi** [comment]
Sorry, I missed that part. Arabic uses phsource/ph_arabic but I'm not sure if that explains it either.

Can you give more examples?

**Reply by cnlinxi** [reply]
Yes. Thanks for your reply.

In dictsource/ar_listx:

1. 
![image](https://github.com/espeak-ng/espeak-ng/assets/11946496/e0f7cb68-3d63-4df0-8b98-c3102e78df16)
```shell
echo "اتحاد" | espeak-ng -qX -v ar
```
output:
```
Found: 'اتحاد' [ittiHaad]  
 ,ittiH'aad
```

2. 
![image](https://github.com/espeak-ng/espeak-ng/assets/11946496/23edee2a-6f11-4151-8b72-b97c4215c837)
```shell
echo "مدينة" | espeak-ng -qX -v ar
```

output:
```
Found: 'مدينة' [madiinah]  
 m,adi'inah
```

from `mad'i:na` to ` m,adi'inah`

3. 
![image](https://github.com/espeak-ng/espeak-ng/assets/11946496/d7450b49-5ab1-47df-b7ee-8aeb98d7f3c6)
```shell
echo "دولار" | espeak-ng -qX -v ar
```

output:
```
Found: 'دولار' [duulaar]  
 d,uul'aar
```

4.
![image](https://github.com/espeak-ng/espeak-ng/assets/11946496/446669b4-8429-43dd-9726-952d552ce022)
```shell
echo "حيث" | espeak-ng -qX -v ar
```

output:
```
Found: 'حيث' [HajTu]  
 H'ajTu
```

This is same.

**Reply by cnlinxi** [reply]
By the way, I would like to know how the phonemes stored in the dictionary are mapped to the final espeak output (IPA)?

For example, which phoneme does `ħ` correspond to in the `dictsource/ar_listx`? 

Thanks a lot.

**Comment by jaacoppi** [comment]
I don't know the details, but you can start with the keyword "ipa" in the phoneme file. 

For example: phsource/ph_arabic has this:
phoneme H       // arabic Hah
  vls phr frc
  ipa ħ
  lengthmod 5
  WAV(vwl_ar/hah,80)
endphoneme

**Reply by cnlinxi** [reply]
Got it.

Thank you👍
